File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        climbs = [cost[0], cost[1]]
        
        if len(cost) > 2:
            for i in range(2, len(cost)):
                prev = climbs[(i-2):i]
                climbs.append(min(prev) + cost[i])
        
        return min(climbs[-1], climbs[-2])

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.swift
class Solution {
    func minCostClimbingStairs(_ cost: [Int]) -> Int {
        var costs = [cost[0], cost[1]]
        
        if cost.count > 2 {
            for i in 2..<cost.count {
                let newCost = cost[i] + min(costs[i-1], costs[i-2])
                costs.append(newCost)
            }
        }
        
        return min(costs[costs.count - 1], costs[costs.count - 2])
    }
}